src/github/github-sync.service.ts

```diff
@@ -457,6 +459,98 @@ export class GithubSyncService {
       await this.removeRepoCollaborator(owner, repo, user, token);
     }
 
+    //sync environments
+    if (FEATURE_FLAG_GITHUB_ENVIRONMENT_SYNC) {
+      const syncableEnvironments = ['development', 'test', 'production'];
+
+      const gitHubEnvironments = await this.listRepoEnvironments(
+        owner,
+        repo,
+        token,
+      );
+
+      const environmentCollection =
+        await this.collectionRepository.getCollections('environment');
+      const filteredEnvironmentCollection = environmentCollection.filter(
+        (env) => {
+          return syncableEnvironments.some(
+            (syncableEnvironment) => syncableEnvironment === env.name,
+          );
+        },
+      );
+
+      for (const index in filteredEnvironmentCollection) {
+        const environmentName = filteredEnvironmentCollection[index].name;
+        const currentEnvironment = gitHubEnvironments.find(
+          (env) => env.name === environmentName,
+        );
+
+        if (currentEnvironment) {
+          await this.removeRepoEnvironment(owner, repo, environmentName, token);
+        }
```

Comment:
Wrap in function call as currentEnvironment lifecycle ends here. 'removeRepoEnvironmentIfExists'